src/NuGet.Clients/PackageManagement.VisualStudio/Migrate/ProjectJsonToPackageRefMigrator.cs

```diff
@@ -22,23 +24,34 @@ public static class ProjectJsonToPackageRefMigrator
             var projectJsonFilePath = ProjectJsonPathUtilities.GetProjectConfigPath(Path.GetDirectoryName(project.MSBuildProjectPath),
                 Path.GetFileNameWithoutExtension(project.MSBuildProjectPath));
 
+            if (!File.Exists(projectJsonFilePath))
+            {
+                throw new FileNotFoundException(string.Format(Strings.Error_FileNotExists, projectJsonFilePath));
+            }
+
             var packageSpec = JsonPackageSpecReader.GetPackageSpec(
                 Path.GetFileNameWithoutExtension(project.MSBuildProjectPath),
                 projectJsonFilePath);
 
-            await MigrateDependencies(project, packageSpec);
+            if (packageSpec == null)
+            {
+                throw new InvalidOperationException(
+                    string.Format(Strings.Error_InvalidJson, projectJsonFilePath));
+            }
+
+            await MigrateDependenciesAsync(project, packageSpec);
 
             var buildProject = EnvDTEProjectUtility.AsMicrosoftBuildEvaluationProject(dteProjectFullName);
```

Comment:
Looks like this will need the UI thread, but I don't see this async method switching to it first.